src/measurements.rs

```diff
@@ -1,21 +1,40 @@
 use ndarray::{s, Array3, ArrayBase, Axis, Data, Ix3, Zip};
+use num_traits::{Bounded, FromPrimitive, Num, NumAssignOps, ToPrimitive, Unsigned};
 
 use crate::Mask;
 
-const BACKGROUND: u16 = 0;
-const FOREGROUND: u16 = 1;
+pub trait LabelType:
+    Copy + Num + FromPrimitive + ToPrimitive + Ord + Unsigned + NumAssignOps + Bounded
```

Comment:
`Unsigned ` already implies `Num`

Hum, I really wish traits in rust were less ugly! I wonder if using [`PrimInt`](https://docs.rs/num-traits/latest/num_traits/int/trait.PrimInt.html) + `Unsigned` would be less ugly. Awww, nope, we would still need `FromPrimitive + ToPrimitive`. I doubt we can make this "better" :(

Reply:
Yeah, with `PrimInt` I can get it down to `PrimInt + FromPrimitive + ToPrimitive + Unsigned + NumAssignOps` (no idea why `NumAssignOps` is not part of `PrimInt`), but it does make the bound more restrictive and don't look that much better IMO.